cg/meta/workflow/fastq.py

```diff
@@ -90,170 +98,119 @@ def get_concatenated_name(linked_fastq_name: str) -> str:
         meta: str | None = None,
     ) -> str:
         """Name a FASTQ file with standard conventions and
-        no naming constrains from pipeline."""
+        no naming constrains from the pipeline."""
         flow_cell: str = f"{flow_cell}-undetermined" if undetermined else flow_cell
         date: str = date if isinstance(date, str) else date.strftime("%y%m%d")
-        return f"{lane}_{date}_{flow_cell}_{sample}_{index}_{read}.fastq.gz"
+        return f"{lane}_{date}_{flow_cell}_{sample}_{index}_{read_direction}{FileExtensions.FASTQ}{FileExtensions.GZIP}"
 
 
 class BalsamicFastqHandler(FastqHandler):
     @staticmethod
     def create_fastq_name(
-        lane: str,
-        flowcell: str,
+        lane: int,
+        flow_cell: str,
         sample: str,
-        read: str,
+        read_direction: int,
         date: dt.datetime = DEFAULT_DATE_STR,
         index: str = DEFAULT_INDEX,
         undetermined: str | None = None,
         meta: str | None = None,
     ) -> str:
         """Name a FASTQ file following Balsamic conventions. Naming must be
         xxx_R_1.fastq.gz and xxx_R_2.fastq.gz"""
-        flowcell = f"{flowcell}-undetermined" if undetermined else flowcell
-        date_str = date if isinstance(date, str) else date.strftime("%y%m%d")
-        return f"{lane}_{date_str}_{flowcell}_{sample}_{index}_R_{read}.fastq.gz"
+        flow_cell = f"{flow_cell}-undetermined" if undetermined else flow_cell
+        date: str = date if isinstance(date, str) else date.strftime("%y%m%d")
+        return f"{lane}_{date}_{flow_cell}_{sample}_{index}_R_{read_direction}{FileExtensions.FASTQ}{FileExtensions.GZIP}"
 
 
 class MipFastqHandler(FastqHandler):
     @staticmethod
     def create_fastq_name(
-        lane: str,
-        flowcell: str,
+        lane: int,
+        flow_cell: str,
         sample: str,
-        read: str,
+        read_direction: int,
         date: dt.datetime = DEFAULT_DATE_STR,
         index: str = DEFAULT_INDEX,
         undetermined: str | None = None,
         meta: str | None = None,
     ) -> str:
         """Name a FASTQ file following MIP conventions."""
-        flowcell = f"{flowcell}-undetermined" if undetermined else flowcell
-        date_str = date if isinstance(date, str) else date.strftime("%y%m%d")
-        return f"{lane}_{date_str}_{flowcell}_{sample}_{index}_{read}.fastq.gz"
+        flow_cell = f"{flow_cell}-undetermined" if undetermined else flow_cell
+        date: str = date if isinstance(date, str) else date.strftime("%y%m%d")
+        return f"{lane}_{date}_{flow_cell}_{sample}_{index}_{read_direction}{FileExtensions.FASTQ}{FileExtensions.GZIP}"
 
 
 class MicrosaltFastqHandler(FastqHandler):
     @staticmethod
     def create_fastq_name(
-        lane: str,
-        flowcell: str,
+        lane: int,
+        flow_cell: str,
         sample: str,
-        read: str,
+        read_direction: int,
         date: dt.datetime = DEFAULT_DATE_STR,
         index: str = DEFAULT_INDEX,
```

Comment:
Are these here just to be an implementation of the FastqHandler method? I am a bit uncertain of what would be best - to let the FastqHandler only hold the the parameters used for all pipelines, and adding the extra parameters where necessary, or to keep it as is. I don't think Python requires implementations of abstract methods to have the same parameters, though. I would probably prefer not having unused parameters as input, but it is just a matter of taste.

Reply:
Another PR